gtsfm/two_view_estimator.py

```diff
@@ -114,19 +162,88 @@ def create_computation_graph(
                 i2Ti1_expected_graph,
                 self._corr_metric_dist_threshold,
             )
+            number_correct, inlier_ratio = corr_error_graph[0], corr_error_graph[1]
         else:
             pose_error_graphs = (None, None)
-            corr_error_graph = None
-
-        return (
-            i2Ri1_graph,
-            i2Ui1_graph,
-            v_corr_idxs_graph,
-            pose_error_graphs[0],
-            pose_error_graphs[1],
-            corr_error_graph,
+            number_correct, inlier_ratio = None, None
+
+        R_error_deg, U_error_deg = pose_error_graphs[0], pose_error_graphs[1]
+
+        two_view_report_graph = dask.delayed(generate_two_view_report)(
+            inlier_ratio_est_model,
+            R_error_deg,
+            U_error_deg,
+            number_correct,
+            inlier_ratio,
+            v_corr_idxs_graph
         )
 
+        result = dask.delayed(self.check_for_degeneracy)(
+            two_view_report_graph, i2Ri1_graph, i2Ui1_graph, v_corr_idxs_graph
+        )
+        i2Ri1_graph, i2Ui1_graph, v_corr_idxs_graph, two_view_report_graph = result[0], result[1], result[2], result[3]
+
+        return (i2Ri1_graph, i2Ui1_graph, v_corr_idxs_graph, two_view_report_graph)
+
+    def check_for_degeneracy(
+        self,
+        two_view_report: TwoViewEstimationReport,
+        i2Ri1: Optional[Rot3],
+        i2Ui1: Optional[Unit3],
+        v_corr_idxs: np.ndarray,
+    ) -> Tuple[Optional[Rot3], Optional[Unit3], np.ndarray]:
+        """ """
+        insufficient_inliers = two_view_report.num_inliers_est_model < self._min_num_inliers_acceptance
+
+        H_EF_inlier_ratio = two_view_report.num_H_inliers / (two_view_report.num_inliers_est_model + EPSILON)
+        is_planar_or_panoramic = H_EF_inlier_ratio > MAX_H_INLIER_RATIO
+
+        # TODO: technically this should almost always be non-zero, just need to move up to earlier
+        valid_model = two_view_report.num_inliers_est_model > 0
```

Comment:
Ideally should we not check for a minimum inlier ratio instead? Something very low like 0.1? If we had 1 inlier, is that a valid model?

Reply:
You also seem to be logging the inlier ratio afterwards not the num inliers.

Reply:
so we are chekcing the minimum inlier ratio also, but that is in the `Verifier` class, not the `TwoViewEstimator` class. I was thinking that belonged in the verifier, since there would be no reason to find a F or E matrix if we already no the inlier ratio w.r.t. est model is too low.

Reply:
I see, okay.